### PR TITLE
More effects on chart player

### PR DESCRIPTION
**Pull request by LiteHell**
- Perodic when doing nothing
- Effect when the note is judged
- Effect for janggu icon at center when it's hitted
- Refactoring

### CODE REVIEW

**Review by firekann — APPROVED** [review]
LGTM

**Review by boulce — CHANGES_REQUESTED** [review]
I don't know it is your intention, at first of the game play, the janggu icon reduces even though user press nothing.

File: game/src/game/game_player/chart_player_ui.rs [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        let background_height_with_border =
+            background_height_without_border + background_border_height * 2;
+
+        // calculate janggu width
+        let janggu_width_min = ((janggu_texture.query().width as f32
+            / janggu_texture.query().height as f32)
+            * background_height_with_border as f32) as u32;
+        let janggu_width_max = janggu_width_min + 20;
+
+        // calc janggu icon size
```
[the remark this answers is not in the document]
Comment (LiteHell) [reply game/src/game/game_player/chart_player_ui.rs:92]: yes

File: game/src/game/game_player/chart_player_ui.rs [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+        }
+
+        // draw disappearing notes, too
+        let note_disappearing_duration = DisapearingNoteEffect::effect_duration();
+        for i in &self.disappearing_note_effects.notes {
+            let tick_delta = i.tick.abs_diff(self.disappearing_note_effects.base_tick);
+            if (matches!(i.note.stick, JangguStick::열채)
+                && tick_delta < note_disappearing_duration)
+            {
+                println!("drawing disappearing note at {}", i.note.distance);
```
[the remark this answers is not in the document]
Comment (LiteHell) [reply game/src/game/game_player/chart_player_ui.rs]: Fixed on 3b8cf9584e771362e471e6177a09660920b32fbc

File: game/src/game/game_player/chart_player_ui.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (LiteHell) [reply game/src/game/game_player/chart_player_ui.rs]: Modified on 4a6d128ebffbe7c45ec574dfef8b8b1c9ac23135

**Comment by LiteHell** [comment]
> I don't know it is your intention, at first of the game play, the janggu icon reduces even though user press nothing.

Fixed on 40811f38b98ff8223e1ef5117a39ac749065a5af

**Review by boulce — CHANGES_REQUESTED** [review]
After start game immediately, if you press just K, it doesn't react, meaning the janggu icon doesn't increase

**Comment by LiteHell** [comment]
> After start game immediately, if you press just K, it doesn't react, meaning the janggu icon doesn't increase

Fixed on 15a70afce5a3c1f33c3f4fd195cc568b35fe3db7

**Review by boulce — APPROVED** [review]
LGTM

**Pull request merged**